src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -2398,53 +2399,115 @@ def delete_ssl_cert(cmd, resource_group_name, certificate_thumbprint):
+            cert_id_parts = parse_resource_id(certificate)
+            cert = cert_id_parts['name']
+            cert_subscription = cert_id_parts['subscription']
+        else:
+            cert = certificate
+    elif key_vault_certificate_name:
+        cert = key_vault_certificate_name
     else:
-        kv_id = key_vault
+        no_cert_msg = 'You must provide a certificate name, use \'--certificate-name\'.'
+        logger.warning(no_cert_msg)
+        return
+
+    (kv_id, kv_secret_name, error) = get_import_kv_details(cmd, client, cert, key_vault, cert_subscription)
+
+    if error:
+        logger.warning(error)
+        return
 
     if kv_id is None:
         kv_msg = 'The Key Vault {0} was not found in the subscription in context. ' \
                  'If your Key Vault is in a different subscription, please specify the full Resource ID: ' \
                  '\naz .. ssl import -n {1} -g {2} --key-vault-certificate-name {3} ' \
                  '--key-vault /subscriptions/[sub id]/resourceGroups/[rg]/providers/Microsoft.KeyVault/' \
-                 'vaults/{0}'.format(key_vault, name, resource_group_name, key_vault_certificate_name)
+                 'vaults/{0}'.format(key_vault, name, resource_group_name, cert)
         logger.warning(kv_msg)
         return
 
     kv_id_parts = parse_resource_id(kv_id)
     kv_name = kv_id_parts['name']
     kv_resource_group_name = kv_id_parts['resource_group']
     kv_subscription = kv_id_parts['subscription']
-    cert_name = '{}-{}-{}'.format(resource_group_name, kv_name, key_vault_certificate_name)
+    if not key_vault:
+        r = random.randrange(0, 99999)
+        cert_name = '{}-{}-{}'.format(cert, resource_group_name, r)
+    else:
+        cert_name = '{}-{}-{}'.format(resource_group_name, kv_name, cert)
     lnk = 'https://azure.github.io/AppService/2016/05/24/Deploying-Azure-Web-App-Certificate-through-Key-Vault.html'
     lnk_msg = 'Find more details here: {}'.format(lnk)
     if not _check_service_principal_permissions(cmd, kv_resource_group_name, kv_name, kv_subscription):
         logger.warning('Unable to verify Key Vault permissions.')
         logger.warning('You may need to grant Microsoft.Azure.WebSites service principal the Secret:Get permission')
         logger.warning(lnk_msg)
 
-    kv_cert_def = Certificate(location=location, key_vault_id=kv_id, password='',
-                              key_vault_secret_name=key_vault_certificate_name, server_farm_id=server_farm_id)
-
+    kv_cert_def = Certificate_Model(location=location, key_vault_id=kv_id, password='',
+                                    key_vault_secret_name=kv_secret_name, server_farm_id=server_farm_id)
     return client.certificates.create_or_update(name=cert_name, resource_group_name=resource_group_name,
                                                 certificate_envelope=kv_cert_def)
 
 
+def get_import_kv_details(cmd, client, certificate_name, key_vault, cert_subscription):
```

Comment:
is this a helper function if so can be private & can be defined with _get_import_*. Also please move helpers to a separate file instead of adding them to custom.py